contentcuration/contentcuration/views/admin.py

```diff
@@ -141,6 +142,149 @@ def get_all_users(request):
+
+    query = None
+    for prop in ['is_admin', 'is_active']:
+        if AND_(prop) in params:
+            q = Q(**{prop: True})
+            q = q if params[AND_(prop)] == 'True' else ~q
+            query = q & query if query else q
+        if OR_(prop) in params:
+            q = Q(**{prop: True})
+            q = q if params[OR_(prop)] == 'True' else ~q
+            query = q | query if query else q
+    
+    if query:
+        user_list = user_list.filter(query)
+
+    if params.get('sort_by') in ['email','fist_name','last_name','channels']:
+        if params.get('ordering') == 'desc':
+            ordering = F(params.get('sort_by')).desc()
+        else:
+            ordering = F(params.get('sort_by')).asc()
+
+        user_list = user_list\
+                    .annotate(channels=Count('editable_channels'))\
+                    .order_by(ordering)
+
+    user_list_page = paginator.paginate_queryset(user_list, request)
+    user_serializer = AdminUserListSerializer(user_list_page, many=True)
+
+    return Response(user_serializer.data)
+
+channel_filters = {
+    'all': Q(),
+    'is_live': ~Q(deleted=True),
+    'is_published': Q(published=True),
+    'is_public': Q(public=True),
+    'is_private': ~Q(public=True),
+    'can_edit': Q(can_edit=True),
+    'is_staged': ~Q(staging_tree=None),
+    'is_ricecooker': ~Q(ricecooker_version=None),
+    'is_deleted': Q(deleted=True)
+}
+
+channel_sortable_keys = [
+    'name',
+    'id',
+    'priority', 
+    'users', 
+    # 'items',      # not working yet!
+    # 'modified',       # not working yet!
+    'created'
+]
+@login_required
+@api_view(['GET'])
+@authentication_classes((SessionAuthentication, BasicAuthentication, TokenAuthentication))
+@permission_classes((IsAdminUser,))
+def get_channels(request):
+    if not request.user.is_admin:
+        raise SuspiciousOperation("You are not authorized to access this endpoint")
+
+    params = request.GET
+    paginator = ChannelUserListPagination()
+    channel_list = Channel.objects.select_related('main_tree').prefetch_related('editors', 'viewers').distinct()    
+
+    if params.get('user_id'):
+        user = User.objects.get(pk=params['user_id'])
+        channel_list = channel_list.filter(editors__in=[user])
+
+    query = None
+    for prop in channel_filters.keys():
+        if AND_(prop) in params:
+            q = channel_filters[prop]
+            q = q if params[AND_(prop)] == 'True' else ~q
+            query = q & query if query else q
+        if OR_(prop) in params:
+            q = channel_filters[prop]
+            q = q if params[OR_(prop)] == 'True' else ~q
+            query = q | query if query else q
+    
+    if query:
+        channel_list = channel_list.filter(query)
+
+    ###### (for reference!) ################
+
+    # 'items'
+    # def get_resource_count(self, channel):
+    #     return channel.main_tree.get_descendants().exclude(kind_id=content_kinds.TOPIC).count()
```

Comment:
This is the `items` count expression I mentioned [here](https://github.com/learningequality/studio/pull/714/files#r193163075).